### ISSUE TITLE
[profiling] unable to load page error when onboarding using version 8.14.0 

### ISSUE THREAD

**Issue by endorama** [elastic/integrations#9845]
This bug has been reported internally, opening a public issue for awareness and tracking resolution.

When opening Universal profiling "Add profiling data" page the user is met with a `Unable to load page error`:
![Screenshot 2024-05-12 at 18 46 15](https://github.com/elastic/integrations/assets/526307/399a0bee-d576-441b-9541-af3c0c4f6fd0)

**Affected integrations:**
- Universal profiling collection, version: 8.14.0

**Affected Kibana versions:** 8.12 up to 8.12.2, 8.13 up to 8.13.4, 8.14 (development)

The reported browser error is [this](https://react.dev/errors/31?invariant=31&args%5B%5D=object%20with%20keys%20%7Bid%2C%20isSecretRef%7D):
```
Objects are not valid as a React child (found: object with keys {id, isSecretRef}). If you meant to render a collection of children, use an array instead.
```

The secret token is rendered as `[object Object]` on the other screens as per: `sudo pf-host-agent-8.13.4-linux-x86_64/pf-host-agent -project-id=1 -secret-token=[object Object] -collection-agent=REDACTED.profiling.europe-west2.gcp.elastic-cloud.com:443`

The root cause is PR #9806 which introduced secrets field in the Universal profiling collector integration.
The cause for this behavior is in the [Onboarding UI](https://github.com/elastic/kibana/blob/551e1232b870654d036aff4d52f43443ee3ecf20/x-pack/plugins/observability_solution/profiling/public/views/add_data_view/index.tsx#L79) which uses that field to display a guided onboarding process for user and is receiving an unexpected data type.

**Comment by trungcle** [comment]
I have a customer who facing this error in version 8.13, is there any workaround or fix yet? or SLA?
thanks

**Comment by endorama** [comment]
@trungcle the fix has been merged and should be available. We have reports of the issue being now fixed.

**Comment by endorama** [comment]
I'm going to close this bug has it has been fixed in integration version 8.14.1.
